distributed/tests/test_worker_client.py

```diff
@@ -297,3 +297,19 @@ def f():
 
     result = yield c.submit(f)
     assert result
+
+
+@gen_cluster()
+def test_submit_different_names(s, a, b):
+    # https://github.com/dask/distributed/issues/2058
+    da = pytest.importorskip('dask.array')
+    c = yield Client('localhost:' + s.address.split(":")[-1], loop=s.loop,
+                     asynchronous=True)
+    try:
+        X = c.persist(da.random.uniform(size=(100, 10), chunks=50))
+        yield wait(X)
```

Comment:
Fine for now, but in the future I recommend `foo.persist()` over `c.persist(foo)`

Reply:
Just for aesthetic reasons really